.circleci/config.yml

```diff
@@ -147,11 +147,6 @@ commands:
       - check-changed-files-or-halt
       - attach_workspace:
           at: '/go'
-      - when:
-          condition:
-            equal: [ windows, << parameters.type >> ]
-          steps:
-            - run: make versioninfo
```

Comment:
@sspaink looking for some advice here

Right now `make versioninfo` is only run once during package builds. Unfortunately, we need to run it again for arm64 windows builds because the goversioninfo command needs to be run with the `-arm` and `-64` flags. Without this, the build will fail.

The versioninfo make target is what ultimately calls go generate on the `cmd/telegraf/telegraf_windows.go` file. Is there a better place to be running the go generate command? Or a better way to inject those flags?

Reply:
The current order of operation is:

1. circle ci calls `make versioninfo`
2. makefile downloads goversioninfo
3. makefile runs go program to generate JSON with product and version in it
4. makefile runs go generate 
5. cirlce ci calls `make package` for each of the 3 builds (i386, amd64, arm64)

We need step 4 to occur before each build. I assume the reason we did not add this to ever build was to avoid having this occur for any user?

Reply:
Right I think the reason to have it be a separate step so that you can continue to build Windows targets without the extra dependency. Wasn't sure who might rely on this. An option could be to have the arm build be under a separate "type" called "windows_arm" or something and have it separate in the makefile. Then we can add a conditional step in here like:

```
      - when:
          condition:
            equal: [ windows_arm, << parameters.type >> ]
          steps:
            - run: make versioninfo_arm
```

Then we can add the extra flags that way? but we could also just add the `make versioninfo` step to run automatically when you run `make package` for windows without it being an extra step.